### DISCUSSION TITLE
Change behaviour of getindex/getproperty for plotobjects

### DISCUSSION THREAD

**Discussion by SimonDanisch** [JuliaPlots/Makie.jl#4212] [Ideas]
To break up https://github.com/JuliaPlots/Makie.jl/pull/1085, I want to start changing the behaviour of getindex/getproperty, without getting into the huge refactor of also changing the planned internal implementation of the plot objects.
The plan is to, change `plot.attribute_name` to:

* return the fully converted value. So `p = plot!(..., color=:red); p.color` will return `RGBAf(...)`. This will help to get rid of lots of Observable indirection and double conversions.
* `p.color = :red` will still work and convert immediately
* if one really needs an observable, one can call `observe(p, :color)` 
* no `p[1]` anymore.. We may want to keep `p.args[1]` for backward compatability, though
* We may need `p.kw_args` to make it possible to access the plot attributes the way the kw arguments allow. E.g., I want to convert colormaps + colorrange to a new object, that will sit at `plot.color` (idea not yet final). So, `p = plot!(..., color=x, colorrange=...); p.color` will not return `x` but instead some kind of color sampler object, which can be annoying. So the proposal would be to allow `p.kw_args.color == x; p.kw_args.color = new_x`, which also opens up the possibility to allow for some magic keyword args.

This all will help a lot with type stability and getting rid of lots of Observable indirections, which are slow to compile and run.
It aims to also make the API cleaner and easier to work with.

**Comment by ffreyer** [comment]
Is this still planned or can this be closed?
